src/middleware.ts

```diff
@@ -37,6 +38,16 @@ const errorHandler: ErrorRequestHandler = (
   if (error instanceof ZodError) {
     res.status(403).json({ error });
   }
+
+  if (error instanceof MongoServerError && error.code === 11000) {
+    res
+      .status(400)
+      .json({
+        success: 'false',
+        message: 'Flashcard with a given front value already exists!',
```

Comment:
**nitpick:** Hmm, domain specific message could be problematic here, as you can also get this error code for other cases when your app will scale